### PR TITLE
File transfer support for leptons

### PR DESCRIPTION
**Pull request by Emmanuel289**
- [x] I have added the tests to cover my changes.
- [x] I have updated the documentation and CHANGELOG accordingly.
- [ ] I have read the CONTRIBUTING document.

### CODE REVIEW

**Comment by AlejandroEsquivel** [comment]
Looking good @Emmanuel289 , you might need the changes from https://github.com/AgnostiqHQ/covalent/pull/742 to make it work however

**Comment by Emmanuel289** [comment]
> Looking good @Emmanuel289 , you might need the changes from #742 to make it work however

Thanks @AlejandroEsquivel. Do I merge #742 directly into the branch or pull the changes after it is merged into develop?

File: tests/covalent_tests/workflow/lepton_test.py [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+                    ],
+                )
+
+    assert mock_c_lepton.language == "C"
+    assert mock_c_lepton.library_name == temp_c_source_file.name
+    assert mock_c_lepton.function_name == "test_entry"
+    assert mock_c_lepton.argtypes == [("c_int", 0), ("LP_c_int", 2), ("LP_c_int", 1)]
+
+
+def test_lepton_languages(
```
Comment (venkatBala) [inline tests/covalent_tests/workflow/lepton_test.py:71]: why do we have the CamelCase and all lower case names for the languages. Is Julia/C++/Fortran also supposed to be on the list?

**Comment by AlejandroEsquivel** [comment]
> > Looking good @Emmanuel289 , you might need the changes from #742 to make it work however
> 
> Thanks @AlejandroEsquivel. Do I merge #742 directly into this branch or pull the changes after it is merged into develop?

@Emmanuel289 You can #741 directly into this branch

**Comment by Emmanuel289** [comment]
> 



> 

@AlejandroEsquivel sounds good, thanks

**Comment by Emmanuel289** [comment]
pre-commit.ci autofix

**Review by AlejandroEsquivel — APPROVED** [review]

File: tests/covalent_tests/workflow/lepton_test.py (hunk 1 above)
Comment (AlejandroEsquivel) [reply tests/covalent_tests/workflow/lepton_test.py:81]: These values are hardcoded in `_LANG_PY` and `_LANG_C` attributes of the Lepton class. They seem to be constants that determine if the supplied `language` kwarg to Lepton class matches their values and effectively makes up the supported languages.

If Julia/C++/Fortran are to be supported then I believe we need more constants in the Lepton class that reference these language but a separate issue for this should be created if that is the case.

**Pull request merged**